library/billing.inc

```diff
@@ -229,40 +253,56 @@ function updateClaim(
         "x12_partner_id = '$partner_id'";
         ****/
         sqlBeginTrans();
-        $version = sqlQuery("SELECT IFNULL(MAX(version),0) + 1 AS increment FROM claims WHERE patient_id = ? AND encounter_id = ?", array( $patient_id, $encounter_id ));
+        $sqlBindArray = array();
+        $sqlBindArray[] = $patient_id;
+        $sqlBindArray[] = $encounter_id;
+        $version = sqlQuery("SELECT IFNULL(MAX(version),0) + 1 AS increment FROM claims WHERE patient_id = ? AND encounter_id = ?", $sqlBindArray);
+
+        $sqlBindArray = array();
+        $sqlBindArray[] = $patient_id;
+        $sqlBindArray[] = $encounter_id;
         if ($crossover<>1) {
             $sql .= "INSERT INTO claims SET " .
-              "patient_id = $patient_id, " .
-              "encounter_id = $encounter_id, " .
+              "patient_id = ?, " .
+              "encounter_id = ?, " .
               "bill_time = NOW() $claimset ," .
               "version = " . $version['increment'];
+            $sqlBindArray = array_merge($sqlBindArray, $sqlBindClaimset);
         } else {//Claim automatic forward case.startTra
             $sql= "INSERT INTO claims SET " .
-              "patient_id = $patient_id, " .
-              "encounter_id = $encounter_id, " .
-              "bill_time = NOW(), status=$status ," .
+              "patient_id = ?, " .
+              "encounter_id = ?, " .
+              "bill_time = NOW(), status=? ," .
               "version = " . $version['increment'];
+            $sqlBindArray[] = $status;
         }
 
-        sqlStatement($sql);
+        sqlStatement($sql, $sqlBindArray);
         sqlCommitTrans();
     } // Otherwise update the existing claim row.
   //
     else if ($claimset) {
+        $sqlBindArray = $sqlBindClaimset;
+        $sqlBindArray[] = $patient_id;
+        $sqlBindArray[] = $encounter_id;
         $claimset = substr($claimset, 2);
         sqlStatement("UPDATE claims SET $claimset WHERE " .
-        "patient_id = '$patient_id' AND encounter_id = '$encounter_id' AND " .
+        "patient_id = ? AND encounter_id = ? AND " .
         // "payer_id = '" . $row['payer_id'] . "' AND " .
-        "version = '" . $row['version'] . "'");
+        "version = '" . $row['version'] . "'", $sqlBindArray);
     }
 
   // Whenever a claim is marked billed, update A/R accordingly.
   //
+    $sqlBindArray = array();
+    $sqlBindArray[] = $payer_type;
+    $sqlBindArray[] = $patient_id;
+    $sqlBindArray[] = $encounter_id;
     if ($status == 2) {
         if ($payer_type > 0) {
             sqlStatement("UPDATE form_encounter SET " .
-            "last_level_billed = '$payer_type' WHERE " .
-            "pid = '$patient_id' AND encounter = '$encounter_id'");
+            "last_level_billed = ? WHERE " .
+            "pid = ? AND encounter = ?", $sqlBindArray);
```

Comment:
This was not necessary. Only need to to $sqlBindArray when you don't know the full query.